scipy/spatial/transform/_rotation.pyx

```diff
@@ -785,6 +785,14 @@ cdef class Rotation:
 
         3D rotations can be represented using unit-norm quaternions [1]_.
 
+        Advanced users may be interested in the "double cover" of 3D space by
+        the quaternion representation [2]_. As of version 1.11.0, the
+        following subset (and only this subset) of operations on a `Rotation`
+        ``r`` corresponding to a quaternion ``q`` are guaranteed to preserve
+        the double cover property: ``r = Rotation.from_quat(q)``,
+        ``r.as_quat(canonical=False)``, ``r.inv()``, and composition using the
+        ``*`` operator such as ``r*r``.
```

Comment:
FWIW, I read through some of the linked open access textbook and the content here is reasonably sensible to me, though I'm certainly not an expert.

Can you clarify for me/remind me what happens for those operations for which application of `-q` doesn't have the same effect as `q`? Is this just "-1 always maps to 1 for redundant cases for supported operations" kind of thing? Since double cover seems to be a fundamental property of quaternions are there "problems" with application of `-q` and `q` for those ops we have that don't support them?

Reply:
Oh, maybe this is more innocent than I think? A bit like preserving IEEE signed zero semantics, where sure you can have `-0.` and `+0`, but really you'll get the same answer for the vast majority of ops so it is kind of a formality? In this case you're saying that there is functionality to prevent canonicalization to the single "positive" representation for the "supported" ops?

Reply:
Yes, we're taking particular care to ensure that we are leaving the quaternions as "native" as we can in those operations and not making unnecessary use of the fact that `q` and `-q` represent the same 3D rotation. For example, to compute the inverse rotation in the quaternion representation, one negates the vector part of the quaternion (hypercomplex conjugation; `conjugate(q)`). Previously, we negated the real part instead (e.g. equivalent to `conjugate(-q)`) and relied on the fact that both of those quaternions represented the same rotation (possibly as a microptimization because only one column needed to get negated). Now, we are implementing `Rotation.inv()` as raw quaternion conjugation. For the most part, if you're not peeking at the quaternion representation, it won't matter, but if you are and doing some fancy things with the quaternion representations (e.g. various kinds of interpolation or dealing with multivalue `Rotation` objects as connected paths), then it can matter.

Reply:
Yeah, agree with all the above. This should have no impact on any operations outside of those specifically using the quaternion double cover in the first place - having both -0 and +0 is a good analogy for the formality here.